contracts/RoyaltyFeeRegistryV2.sol

```diff
@@ -0,0 +1,189 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {OwnableUpgradeable} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import {Initializable} from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+
+import {IRoyaltyFeeRegistryV2} from "./interfaces/IRoyaltyFeeRegistryV2.sol";
+import {RoyaltyFeeTypes} from "./libraries/RoyaltyFeeTypes.sol";
+
+error RoyaltyFeeRegistryV2__InvalidMaxNumRecipients();
+error RoyaltyFeeRegistryV2__RoyaltyFeeCannotBeZero();
+error RoyaltyFeeRegistryV2__RoyaltyFeeLimitTooHigh();
+error RoyaltyFeeRegistryV2__RoyaltyFeeRecipientCannotBeNullAddr();
+error RoyaltyFeeRegistryV2__RoyaltyFeeSetterCannotBeNullAddr();
+error RoyaltyFeeRegistryV2__RoyaltyFeeTooHigh();
+error RoyaltyFeeRegistryV2__TooManyFeeRecipients();
+
+/**
+ * @title RoyaltyFeeRegistryV2
+ * @notice It is a royalty fee registry for the Joepeg exchange and auction house.
+ */
+contract RoyaltyFeeRegistryV2 is
+    IRoyaltyFeeRegistryV2,
+    Initializable,
+    OwnableUpgradeable
+{
+    using RoyaltyFeeTypes for RoyaltyFeeTypes.FeeInfoPart;
+
+    /// @notice Max royalty fee bp allowed (10,000 = 100%)
+    uint256 public royaltyFeeLimit;
+
+    /// @notice Max number of royalty fee recipients allowed
+    uint8 public maxNumRecipients;
+
+    /// @notice Stores royalty fee information for collections
+    mapping(address => RoyaltyFeeTypes.FeeInfoPart[])
+        public royaltyFeeInfoPartsCollection;
+
+    /// @notice Stores setter address for collections whose royalty fee information
+    /// are overridden
+    mapping(address => address) public royaltyFeeInfoPartsCollectionSetter;
+
+    event RoyaltyFeeLimitSet(
+        uint256 oldRoyaltyFeeLimit,
+        uint256 newRoyaltyFeeLimit
+    );
+    event MaxNumRecipientsSet(
+        uint256 oldMaxNumRecipients,
+        uint256 newMaxNumRecipients
+    );
+    event RoyaltyFeeInfoSet(
+        address indexed collection,
+        address indexed setter,
+        RoyaltyFeeTypes.FeeInfoPart[] feeInfoParts
+    );
+
+    modifier isValidRoyaltyFeeLimit(uint256 _royaltyFeeLimit) {
+        if (_royaltyFeeLimit > 9500) {
+            revert RoyaltyFeeRegistryV2__RoyaltyFeeLimitTooHigh();
+        }
+        _;
+    }
+
+    modifier isValidMaxNumRecipients(uint256 _maxNumRecipients) {
+        if (_maxNumRecipients == 0) {
+            revert RoyaltyFeeRegistryV2__InvalidMaxNumRecipients();
+        }
+        _;
+    }
+
+    /**
+     * @notice Initializer
+     * @param _royaltyFeeLimit new royalty fee limit (500 = 5%, 1,000 = 10%)
+     * @param _maxNumRecipients new maximum number of royalty fee recipients allowed
+     */
+    function initialize(uint256 _royaltyFeeLimit, uint8 _maxNumRecipients)
+        public
+        initializer
+        isValidRoyaltyFeeLimit(_royaltyFeeLimit)
+        isValidMaxNumRecipients(_maxNumRecipients)
+    {
+        __Ownable_init();
```

Comment:
Don't forget to transfer the ownership to the `royaltyFeeSetterV2` after the deployment.